pkg/document/crdt/root.go

```diff
@@ -76,15 +76,41 @@ func (r *Root) FindByCreatedAt(createdAt *time.Ticket) Element {
 }
 
 // RegisterElement registers the given element to hash table.
-func (r *Root) RegisterElement(elem Element) {
-	r.elementMapByCreatedAt[elem.CreatedAt().Key()] = elem
+func (r *Root) RegisterElement(element Element) {
```

Comment:
Remove the RegisterElement call in Descendants.
It will be conducted by RegisterElement above.

+++

Fixed RegisterElement recursive call
Just make it call below code rather than recursive call.
```
r.elementMapByCreatedAt[elem.CreatedAt().Key()] = elem
```
The recursive call make duplicate register when set nested object or array.

https://github.com/yorkie-team/yorkie/blob/0cc5001954868845916275db5dd3075b0743ba99/pkg/document/crdt/root.go#L78-L91